src/test/functionDetails.test.ts

```diff
@@ -0,0 +1,275 @@
+  }`;
+  const funcIterator = iterFunctions(code, "Go");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "Go")).toBe("outer");
+});
+// C Tests
+test("C: function_definition", () => {
+  const code = `static void
+tr_where (const void caller, Dl_infoinfo)
+{
+  if (caller != NULL)
+  {
+    printf ("tr_where: %s\n", caller->tr_name);  
+  }
+  else
+  {
+    printf ("tr_where: NULL\n");
+  }
+}`;
+  const funcIterator = iterFunctions(code, "C");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "C")).toBe("tr_where");
+});
+test("C: inline function", () => {
+  const code = `inline int add(int a, int b) {
+    return a + b;
+  }`;
+  const funcIterator = iterFunctions(code, "C");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "C")).toBe("add");
+});
+// C++ Tests
+test("C++: function_definition", () => {
+  const code = `unsigned int Miner::calculate_hash_code() {
+    timestamp = std::time(nullptr); //take the current time since 1970 (unix epoch)
+        std::string data_to_hash = std::to_string(height_target) + std::to_string(nonce) + std::to_string(timestamp) +
+                                   std::to_string(last_hash) + std::to_string(id);
+    uLong crc_res = 0;
+    crc_res = crc32(crc_res, reinterpret_cast<const Bytef *>(data_to_hash.c_str()), data_to_hash.size());
+    //reinterpret cast is used to cast the pointer.
+    return crc_res;
+}`;
+  const funcIterator = iterFunctions(code, "C++");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "C++")).toBe("calculate_hash_code");
+});
+test("C++: lambda_expression - variable", () => {
+  const code = `std::function<int(int)> func2 = [&](int value) -> int {
+        return 1+value+important;
+    };`;
+  const funcIterator = iterFunctions(code, "C++");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "C++")).toBe("func2");
+});
+test("C++: lambda_expression - Anonymous", () => {
+  const code = `[](int value) {
+    std::cout << value << std::endl;
+};`;
+  const funcIterator = iterFunctions(code, "C++");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "C++")).toBe("Anonymous");
+});
+test("C++: template function", () => {
+  const code = `template <typename T>
+T add(T a, T b) {
+  return a + b;
+}`;
+  const funcIterator = iterFunctions(code, "C++");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "C++")).toBe("add");
+});
+// Python Tests
+test("Python: function_definition", () => {
+  const code = `def now(tz: Optional[TZ_EXPR] = None) -> Arrow:
+    """Calls the default :class:ArrowFactory <arrow.factory.ArrowFactory> now method."""
+    return _factory.now(tz)`;
+  const funcIterator = iterFunctions(code, "Python");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "Python")).toBe("now");
+});
+test("Python: function_definition", () => {
```

Comment:
Please avoid giving multiple tests the same description.
If they are testing the same thing it's pointless duplication.
If they are testing different things then the description is not good enough.